descriptors/ModuleDescriptor-template.json

```
@@ -40,7 +40,33 @@
             "inventory-storage.instances.collection.get",
             "inventory-storage.instances.item.get"
           ]
-        }, {
+        },
+        {
+          "methods": [
+            "POST"
+          ],
+          "pathPattern": "/inventory/items/{id}/mark-withdrawn",
```

Could this be shortened to `withdraw`, is the marking aspect significant?

```suggestion
          "pathPattern": "/inventory/items/{id}/withdraw",
```

---

I just want to have a general name convention. In this case an endpoint to mark an item missing will be `/inventory/items/{id}/miss|lose`?

---

> I just want to have a general name convention. In this case an endpoint to mark an item missing will be `/inventory/items/{id}/miss|lose`?

Interesting. I think that might be challenging. Could you expand a little on what you are thinking of as a convention, maybe with examples of other endpoints?

I think the example of an item being lost vs. withdrawn is especially interesting.

I think that withdrawing an item, is something that a library staff member actively decides to do. Whereas I think an item being lost is an event that has already occurred, and the staff member is changing the system to catch up with reality.

I think this is a domain modelling challenge as much as it is an API design challenge. My thinking was primarily on trying to succinctly express the domain intent.

---

I think endpoints can be named like `/mark-{target-status}`, e.g. `/mark-missing`, `/mark-withdrawn`, etc.

Please note, that UI changes has been already merged, it will require a follow up PR for UI.

---

Yeah, it's too late to change this API design at this point.

I think we need to be careful not to try too hard to fit these into the same naming pattern, I think that might be a missed opportunity for modelling these processes rather than only as status changes.

Anyway, I'm resolving this because I gave this feedback too late :-(